Deprecate storage-type flag

jimmidyson opened the pull request:
This uses the same conventions as Kubernetes, not marking flag as deprecated via pflag.FlagSet because this hides the flag from help, but rather marking the usage message as DEPRECATED.

As discussed in #2201, this keeps the flag and chart values to not break existing deployments, but completely unwires it so it can be removed in a (near-)future release.

jimmidyson commented:
/assign @carolynvs

MHBauer commented:
why do we want to show the flag in help? to act as notification that it is deprecated?

jimmidyson commented:
I've refactored this following https://github.com/spf13/pflag/pull/163 to use the "proper" `MarkDeprecated` function but also ensuring that it's visible in usage - this makes it more visible that it will be unused and removed in the next few releases.

MHBauer commented on cmd/apiserver/app/server/options.go:
```diff
@@ -23,23 +23,20 @@ import (
 	certDirectory = "/var/run/kubernetes-service-catalog"
 
-	storageTypeFlagName = "storageType"
+	storageTypeFlagName = "storage-type"
```
this doesn't change the flag, as far as I can tell. We just weren't using it before.

jimmidyson replied:
It wasn't being used before, but it is now - see https://github.com/kubernetes-incubator/service-catalog/pull/2266/files#diff-5dd3f709c02872b9414e257945813685R89 and following lines.

jimmidyson commented:
@carolynvs Any chance of a review of this?

carolynvs commented on cmd/apiserver/app/server/options.go:
```diff
@@ -86,12 +83,15 @@ func NewServiceCatalogServerOptions() *ServiceCatalogServerOptions {
 		"The type of backing storage this API server should use",
 	)
+	flags.MarkDeprecated(storageTypeFlagName, "The value of this flag is now unused and will be removed in the near future")
+	flags.Lookup(storageTypeFlagName).Hidden = false
```
👍

jimmidyson commented:
@carolynvs Thanks for the review! Is there anyone else that can review so we could get this merged? Thanks!

carolynvs commented:
/assign @jboyd01 

Jay, I'm hoping you have time to give this a lgtm? 😀

jboyd01 commented:
sorry this has taken so long to get merged @jimmidyson, we appreciate the contribution!
/lgtm
